packages/twenty-front/src/modules/ui/utilities/hotkey/hooks/useGoToHotkeys.ts

```diff
@@ -5,13 +5,24 @@ import { AppHotkeyScope } from '../types/AppHotkeyScope';
 
 import { useSequenceHotkeys } from './useSequenceScopedHotkeys';
 
-export const useGoToHotkeys = (key: Keys, location: string) => {
+type GoToHotkeysProps = {
+  key: Keys;
+  location: string;
+  preNavigateFunction?: () => void;
+};
+
+export const useGoToHotkeys = ({
+  key,
+  location,
+  preNavigateFunction,
```

Comment:
passing a a preNavigationFunction to update the problematic state

Reply:
Could you please explain me how this part is working?

I mean we did avoid useEffect but how's it solving the problem?